packages/contracts/src/EmailWalletCore.sol

```diff
@@ -99,6 +100,11 @@ contract EmailWalletCore is ReentrancyGuard, OwnableUpgradeable, UUPSUpgradeable
     // Gas required to claim unclaimed state
     uint256 public immutable unclaimedStateClaimGas;
 
+    // Sora's comment: this value is used in `registerUnclaimedState` to set a max gas for calling `extension.registerUnclaimedState`.
+    // Gas required to register unclaimed state
+    uint256 public immutable unclaimedStateRegisterGas;
```

Comment:
> But this is not limited to unclaimed states - now that extensions can make execute() on wallets, they could call other contracts that consume a lot of gas and gain using the same attack above.

Since the extension call execute() on wallets when the user calls the extension or the unclaimed state is claimed/reverted, I think its gas already is capped by maxGasOfExtension[extension] or unclaimedStateClaimGas.

Reply:
There is a maxGas for exctension.execute which is the only place that invoke registerUnclaimedStateAsExtension. The max gas for the whole operation execution is capped at extension.maxGas.
For EOA registering unclaimed state the wallet ui will show gas and set maxGas.
In this case, why do we need `unclaimedStateRegisterGas`?

Reply:
> For EOA registering unclaimed state the wallet ui will show gas and set maxGas.

Sorry, I forgot that.
When the external EOA/contract registers an unclaimed state, it should understand the spec of `registerUnclaimedState` to be called.
Therefore, the core does not need to set its max gas (`unclaimedStateRegisterGas `).

Reply:
Yes right. Both external contracts and EOA can set the gas limit when calling the extension. 
I will revert this change.